### ISSUE TITLE
Kodi crashes while reconnecting to TVheadend

### ISSUE THREAD

**Issue by Aydan** [xbmc/xbmc#16311]
### Describe the bug
Here is a clear and concise description of what the problem is:

After the update from 18.2 to 18.3 i had crashes immediately after startup, which I could isolate to the connection with TVheadend.
kodi-pvr-hts has not been updated during the update to 18.3
I'm using the deb-multimedia stretch-backports repo.
The problem occurred immediately after starting the updated Kodi.
After renaming ~/.kodi, it was OK for a few days, now it just happened again.
Problem reoccurred after resume from hibernation, but it was not the first resume from hibernation with the new profile.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Normal startup and reconnection with TVheadend.

## Actual Behavior

Kodi crashes, as soon as network connection to TVheadend is established

## Possible Fix

Temporary workaround:
reset kodi config.

### To Reproduce

May be hard to reproduce, but I could provide my crashing profile.
1. Unplug network (TVheadend must be on another computer)
2. Start Kodi
3. Plugin network
4. Kodi crashes

### Debuglog

The debuglog can be found here:
[crash at startup](https://pastebin.com/ZE2v1Ubi)
[crash after plugging in network](https://pastebin.com/1VaHbr5t)

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
Debian Stretch with backports kernel
 - Kodi version:
18.3-dmo1~bpo9+1 from deb-multimedia stretch-backports
```
kodi/stable-backports,stable-backports,now 5:18.3-dmo1~bpo9+1 all  [installiert]
kodi-bin/stable-backports,now 5:18.3-dmo1~bpo9+1 amd64  [Installiert,automatisch]
kodi-data/stable-backports,stable-backports,now 5:18.3-dmo1~bpo9+1 all  [Installiert,automatisch]
kodi-inputstream-adaptive/stable-backports,now 2.3.18-dmo1~bpo9+1 amd64  [installiert]
kodi-inputstream-rtmp/stable,now 1.0.5-dmo1+deb9u1 amd64  [installiert]
kodi-pvr-hts/stable-backports,now 6:4.4.18-dmo1~bpo9+1 amd64  [installiert]

```

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by ksooo** [comment]
At a first glance, it seems your video database is damaged:

<pre>
2019-06-26 20:57:39.230 T:139772966680320   ERROR: SQL: Field not found: path.idPath
2019-06-26 20:57:39.230 T:139772966680320   ERROR: GetPathId unable to getpath (select idPath from path where strPath='pvr://recordings/tv/active/')
2019-06-26 20:57:39.230 T:139772979963648   ERROR: SQL: Field not found: path.idPath
2019-06-26 20:57:39.230 T:139772966680320   ERROR: SQL: Field index not found: -1
2019-06-26 20:57:39.230 T:139772979963648   ERROR: GetPathId unable to getpath (select idPath from path where strPath='pvr://recordings/tv/active/')
2019-06-26 20:57:39.230 T:139772979963648   ERROR: Previous line repeats 1 times.
2019-06-26 20:57:39.230 T:139772979963648   ERROR: SQL: Field not found: path.idPath
2019-06-26 20:57:39.230 T:139772979963648   ERROR: GetPathId unable to getpath (select idPath from path where strPath='pvr://recordings/tv/active/')
2019-06-26 20:57:39.231 T:139772979963648   ERROR: SQL: Field index not found: -1
2019-06-26 20:57:39.231 T:139772979963648   ERROR: GetPathId unable to getpath (select idPath from path where strPath='pvr://recordings/tv/active/')
2019-06-26 20:57:39.231 T:139772979963648   ERROR: SQL: Field not found: path.idPath
2019-06-26 20:57:39.231 T:139772979963648   ERROR: GetPathId unable to getpath (select idPath from path where strPath='pvr://recordings/tv/active/')
2019-06-26 20:57:39.231 T:139772979963648   ERROR: SQL: Field not found: files.idFile
2019-06-26 20:57:39.231 T:139772979963648   ERROR: GetFileId (pvr://recordings/tv/active/Hawaii%20Five-0%20Das%20letzte%20Kapitel%2c%20Crime-Serie%2c%20USA%202018, TV%20(SAT.1), 20190409_200717, 1923646254.pvr) failed
</pre>

**Comment by ksooo** [comment]
Second glance reveals that we're facing an issue with concurrent access to CVideoDatabase. Very ugly.

**Comment by Aydan** [comment]
So there's a thread lock missing in the sql access driver?

**Comment by a1rwulf** [comment]
> Second glance reveals that we're facing an issue with concurrent access to `CVideoDatabase` - T1 vs T20 in first log, T3 vs T1 in second log. Very ugly.

Afaik sqlite should handle that: https://www.sqlite.org/threadsafe.html
We also have a lot of potential concurrent accesses to `CVideoDatabase` if you fire `VideoLibrary.GetMovies` JSON requests from multiple clients.

**Comment by Aydan** [comment]
is there a way to know if this commit has made it into 18.3-dmo1~bpo9+2?

**Comment by a1rwulf** [comment]
> is there a way to know if this commit has made it into 18.3-dmo1~bpo9+2?

It's not in 18.3.

**Comment by arnova** [comment]
> > is there a way to know if this commit has made it into 18.3-dmo1~bpo9+2?
> 
> It's not in 18.3.

But it will be in 18.4, if we decide to release it.

**Comment by Aydan** [comment]
OK, that sucks, because this bug has destroyed my profile 5 times in the last weeks., and I'm fed up with losing all my settings and playback pointers all the time.

**Comment by ksooo** [comment]
You could go back to 18.2

**Comment by Aydan** [comment]
> > > is there a way to know if this commit has made it into 18.3-dmo1~bpo9+2?
> > 
> > 
> > It's not in 18.3.
> 
> But it will be in 18.4, if we decide to release it.

I hope you meant "when", not "if".
Is there any date set for the next release?
I'm asking because Christian Marillat from deb-multimedia didn't want to do a bugfix release for 18.3 if 18.4 was around the corner.

Regards
Aydan

**Comment by arnova** [comment]
> > > > is there a way to know if this commit has made it into 18.3-dmo1~bpo9+2?
> > > 
> > > 
> > > It's not in 18.3.
> > 
> > 
> > But it will be in 18.4, if we decide to release it.
> 
> I hope you meant "when", not "if".
> Is there any date set for the next release?
> I'm asking because Christian Marillat from deb-multimedia didn't want to do a bugfix release for 18.3 if 18.4 was around the corner.
> 
> Regards
> Aydan

Like I said: it hasn't been decided yet whether there will be a 18.4 release.

**Comment by Aydan** [comment]
Just to confirm: Is this fix in the recently released 18.4? Otherwise I'l stay with 18.2

**Comment by ksooo** [comment]
> Is this fix in the recently released 18.4? 

 It is.
